ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1279,6 +1296,35 @@ knownIntersectionStateTop cfgEnv dynEnv intEnv =
                   , kBestBlockNo = max (fromWithOrigin 0 $ AF.headBlockNo $ jTheirFragment ji) (kBestBlockNo kis)
                   }
 
+          combineJumpFragment ::
+               KnownIntersectionState blk
+            -> AnchoredFragment (Header blk)
+            -> KnownIntersectionState blk
+          combineJumpFragment kis@KnownIntersectionState{ourFrag} fragment =
+            let mRewoundHistory =
+                  HeaderStateHistory.rewind
+                    (AF.castPoint $ AF.headPoint fragment)
+                    (theirHeaderStateHistory kis)
+                rewoundHistory =
+                  -- When jumping to a good point, the history should be
+                  -- rewindable because it was rewindable when the good point
+                  -- was discovered.
+                  fromMaybe (error "offerJump: cannot rewind history for a good point") mRewoundHistory
+                intersection =
+                  case AF.intersect ourFrag fragment of
+                    Just (po, _, _, _) -> castPoint $ AF.headPoint po
+                    -- ourFrag should intersect with the good fragment, or otherwise
+                    -- it would not intersect with it at the time the good
+                    -- fragment was discovered.
+                    Nothing -> error "offerJump: the fragment should have a valid intersection with the current selection"
```

Comment:
@amesgen, @niols, attention here.

I believe the error calls in this function are unreachable, but this is not trivial to see. Perhaps the comments are easier to grasp when assuming the dynamo doesn't change. But when the dynamo changes, the last good point is assumed to be the anchor of the dynamo's candidate fragment (i.e. it hasn't actually been sent before via `MsgFindIntersect`).

Reply:
This seems fine to me when the dynamo does not change (couldn't we "freeze" it as soon as someone objected?), as we are in this scenario:

 - The dynamo offers a jump to all jumpers.
 - Someone objects (suppose just one jumper); we find the intersection with the dynamo.
 - Hence, when spinning up that objector, we want to set its `KnownIntersectionState` such that its `theirFrag objector` ends the computed intersection. So `theirFrag objector` will be a prefix of `theirFrag dynamo`, and as `theirFrag dynamo` intersects with `ourFrag dynamo`, so does `theirFrag objector`.

Minor: This code could use `AF.intersectionPoint` instead of `AF.intersect`.

Reply:
Adopted `intersectionPoint`. Iterated once more on the implementation to use a kis from the dynamo instead of working with the kis from the objector to set the intersection of the ChainSync server.